pkg/controller/azvolume.go

```diff
@@ -82,11 +82,11 @@ func (r *ReconcileAzVolume) Reconcile(ctx context.Context, request reconcile.Req
 		return reconcile.Result{Requeue: true}, err
 	}
 
-	if now := metav1.Now(); azVolume.ObjectMeta.DeletionTimestamp.Before(&now) {
+	if !azVolume.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
This shouldn't be an issue as of now because k8s does not support graceful deletion for CRIs. But I think we should use `.Before()` just so that we don't have to revert these changes once the community does decide to support graceful deletion. It doesn't hurt to use `.Before` anyways.

Reply:
Oh is there a plan to support this in the community? I changed this because as far as I could tell, this was the recommended approach in the community. How is the graceful deletion going to work?

Reply:
I know that there has been a few discussion revolving this in the past few years but I don't think they have one going right now. So I am not certain what it would look like. But if that does happen, I would presume the deletionTimestamp for CRs would be allowed to be modified via setting graceperiod option upon deletion request.
https://github.com/kubernetes/kubernetes/blob/9d38c64a7da5d4c7688ba11334485b8cabd9dc1c/staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go#L150-L152